src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_UpdateSource.cs

```diff
@@ -2,17 +2,20 @@
 
 using System;
 using System.Collections.Immutable;
-using System.Composition;
 using System.Threading;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
-    [Export(typeof(IDiagnosticUpdateSource))]
     internal partial class DiagnosticAnalyzerService : IDiagnosticUpdateSource
     {
         public event EventHandler<DiagnosticsUpdatedArgs> DiagnosticsUpdated;
 
+        public DiagnosticAnalyzerService(IDiagnosticUpdateSourceRegistrationService registrationService) : this()
```

Comment:
this is just one of constructor. actual constructor that MEF will call has the attribute.